Makefile

```diff
@@ -119,13 +119,15 @@ init_upstream:
 			(cd upstream && git submodule update --init && git remote add source git@github.com:hashicorp/terraform-provider-aws.git) ; \
 		fi; \
 
-patch_upstream:
-	cd upstream-tools && \
-		yarn install --frozen-lockfile --silent
-	cd upstream && git checkout . && cd - || exit
-	cd upstream-tools && \
-		yarn --silent run tf-patch apply --cwd ../upstream && \
-		yarn --silent run tf-patch check --cwd ../upstream
+patch_upstream: init_upstream
```

Comment:
Don't you need to `cd ..` in between the steps? I don't know make's intricacies like you do though :)

Reply:
Make runs every command within a target in an isolated shell by default. This can be turned off in newer versions (4.x) using `.ONESHELL` I think, but we're still supporting 3.x as that's what macos ships with so we're stuck with either chaining everything into a single line with `&&` or just cd'ing before each command.